lib/utils/fs.go

```diff
@@ -84,27 +85,59 @@ func IsDir(path string) bool {
 
 // NormalizePath normalises path, evaluating symlinks and converting local
 // paths to absolute
-func NormalizePath(path string) (string, error) {
+func NormalizePath(path string, evaluateSymlinks bool) (string, error) {
 	s, err := filepath.Abs(path)
 	if err != nil {
 		return "", trace.ConvertSystemError(err)
 	}
-	abs, err := filepath.EvalSymlinks(s)
-	if err != nil {
-		return "", trace.ConvertSystemError(err)
+	if evaluateSymlinks {
+		s, err = filepath.EvalSymlinks(s)
+		if err != nil {
+			return "", trace.ConvertSystemError(err)
+		}
 	}
-	return abs, nil
+	return s, nil
+}
+
+// OpenFileAllowingSymlinks will open a file, if it's a symlink the file referenced by the link will be associated to
+// the returned os.File.  This will return an error if the file is not found or is a directory.
+func OpenFileAllowingSymlinks(path string) (*os.File, error) {
+	return openFile(path, true)
+}
+
+// OpenFileNoSymlinks will open a file, ensuring it's an actual file and not a directory or symlink.
+func OpenFileNoSymlinks(path string) (*os.File, error) {
+	return openFile(path, false)
 }
 
-// OpenFile opens  file and returns file handle
-func OpenFile(path string) (*os.File, error) {
-	newPath, err := NormalizePath(path)
+// openFile opens a file and returns file handle. In general symlinks should not be allowed to reduce the risk of
+// privilege escalation from Teleports elevated privileges to potentially less privileged users accidentally.
```

Comment:
I think we can kill this comment, the public methods should cover all of this.

```suggestion
```

Reply:
Removed in a216f6d14848719cd061eb29b1ed657247eaee25